### DISCUSSION TITLE
Welcome to EOD2 Discussions!

### DISCUSSION THREAD

**Discussion by BennyThadikaran** [BennyThadikaran/eod2#54] [Announcements]
## 👋 Welcome!

Hi, I'm Benny Thadikaran. Looking forward to connecting with our community members! I hope that you:
  * Ask questions.
  * Share ideas.
  * Engage with others to build our community.  💪

Tell us in the comments below how you use EOD2.
I will also answer any questions related to this repository.

**Comment by ArunTS96** [comment]
Great work, Benny! Your efforts have truly inspired me. I'm currently working on creating a Python module to fetch stock information from NSE, including market capitalization, free float, and more. Your work has been a guiding light in this endeavor. Congratulations once again, and please keep up the fantastic work!

**Reply by BennyThadikaran** [reply]
Thanks @ArunTS96 for your kind words. Leave a link here when your module is uploaded. I'd love to learn any new ideas and maybe contribute to the code.

**Reply by ArunTS96** [reply]
@BennyThadikaran Most of what I tried to do has been already achieved by you here - https://github.com/BennyThadikaran/NseIndiaApi/blob/main/src/samples/quote.json

Only thing that is missing is free float market cap, it would be great if that become part of quote.json.

I will focus on a module that uses your eod2 and NseIndiaApi to rank leading sectors.

**Comment by BennyThadikaran** [comment]
> Only thing that is missing is free float market cap, it would be great if that become part of quote.json.

I have not clearly documented it. The [`NSE.quote` method](https://bennythadikaran.github.io/NseIndiaApi/usage.html#nse.NSE.quote) has a `section` argument set to None by default. If you pass instead `trade_info` it returns the Market cap info. 

![Screenshot from 2023-11-19 15-23-29](https://github.com/BennyThadikaran/eod2/assets/19973418/f7369164-228d-41dc-b87d-c026019c0763)

In the screenshot example you can access it : `data['marketDeptOrderBook']['tradeInfo']['totalMarketCap']`

I will update the documentation soon so its more clear.

> I will focus on a module that uses your eod2 and NseIndiaApi to rank leading sectors.

That would be awesome. I had some ideas for market breadth indicators on EOD2. Perhaps i'll post it on the discusions this week.

**Reply by ArunTS96** [reply]
Awesome. Waiting for the discussion. What I wanted to acheive was market smith style industry ranking and following by which industry/sector is raising in ranks on weekly and monthly time frame.

**Comment by rsjain51** [comment]
Amazing, really great.
I am new to python and coding. faced problem using in start to run "py init.py" but worked after using python3 indtead py
I want to download or add nse del and bhav data for previous days as it only fetched data from 8apr to 16apr only.
And sir can i connext to you on your telegram t.me/BobDoe

Thank you

**Reply by BennyThadikaran** [reply]
Hi rsjain51, 

You can check the [installation instructions here](https://github.com/BennyThadikaran/eod2/wiki/Installation). Read it carefully as there are important details. You can also read the [WIKI](https://github.com/BennyThadikaran/eod2/wiki) on how to use the script. 

For linux and Mac users, its `python3` on windows its `py`. It good you figured this out. :thumbsup:

> as it only fetched data from 8apr to 16apr only.

If you followed the instructions correctly, you can open any file in `src/eod2_data/daily` and it should contain all the data you need.

If you need all bhavcopies just reach out to me on Telegram. I can share them with you. Perhaps i will upload the data and provide a link to download it for other users as well

> can i connext to you on your telegram t.me/BobDoe

Absolutely it would be my pleasure to connect with you!

**Comment by Prady04** [comment]
@BennyThadikaran a feature request 
can there be a generic mechanism for saving data that I can extend to write to other datalakes?

**Reply by BennyThadikaran** [reply]
This seems like a good idea. I had previously considered parquet and other formats for its storage benefits. But it would be too much of a breaking change for most users.

This is just a quick idea that came to mind.

- Define an abstract class with methods that must be defined for the saving behaviour. 
- Write a class for default saving behaviour (saving as CSV files).
- Provide a plugin mechanism so user can define his own saving mechanism.

Maybe i am over complicating a simple task. :smiley_cat:  Do you have something in mind? I dont know much about AWS or what is involved. 

I can create a separate thread to discuss the implementation so it benefits you.

**Comment by Prady04** [comment]
Sure on the new thread. Almost all databases can read csvs into tables. So
your idea of plugin  or api kind of access makes perfect sense to me.
As you are defining the plugin/ api interfaces it's on the implementor to
figure out the details wrt to their storage.

regards,
Pradeep

**Reply by BennyThadikaran** [reply]
I have created an Feature request #130 in EOD2 issues. 

I had created a [plugin system](https://github.com/BennyThadikaran/eod2/blob/main/src/defs/Plugin.py) though I havent done much work on it. I can implement something similar or maybe there is a better way. 

Give me a day or two, I will go through the code and figure out an interface that we can work with. Going forward we can discuss directly in GH issue.

**Reply by BennyThadikaran** [reply]
Hi Pradeep, I have responded to the issue. Check #130

**Comment by cotton1234** [comment]
Hello Benny,

I was checking the NSE website as today it did the download the bhavcopy and I see the following 

To be discontinued w.e.f July 08,2024. Refer NSE Circular No. 62424 dated June 12, 2024. Switch to CM-UDiFF Common Bhavcopy Final(zip)

Thanks
Saurabh

**Reply by BennyThadikaran** [reply]
Hey thanks for bringing this up. Gave me enough heads up to start effecting changes.

**Reply by BennyThadikaran** [reply]
There is also this change in tick size for NSE stocks below Rs. 250 to 0.01 from 0.05. The last day of the previous month will be used as basis. The problem is when making adjustments for splits and bonus, i normally round the entire price history to nearest 0.05. Now what would be the rounding factor 0.01 or 0.05? :stuck_out_tongue: 

https://zerodha.com/marketintel/bulletin/380460/change-in-tick-size-of-nse-stocks-and-fo-contracts-trading-below-250

**Reply by cotton1234** [reply]
I was not aware of the tick size. thanks for letting me know.
